sql-parser/src/main/java/io/crate/sql/Literals.java

```diff
@@ -172,4 +172,29 @@ private static int calculateMaxCharsInSequence(CharSequence seq,
         }
         return idx;
     }
+
+    /*
+       This is a striped down version of Integer::parseInt(CharSequence s, int beginIndex, int endIndex, int radix)
+       introduced in java9 - required here for backward compatibility with java8
+       todo: remove this and replace calls with Integer::parseInt
+       once backward compatibility with java8 in no longer required
+     */
+    private static int parseInt(CharSequence s, int beginIndex, int endIndex, int radix) {
+        assert (s != null) : "input char sequence cannot be null";
+        assert (radix >= Character.MIN_RADIX && radix <= Character.MAX_RADIX) : "radix needs to have a valid value";
+        assert (endIndex > 0) : "endIndex needs to be greater than 0";
+        assert (beginIndex >= 0 && beginIndex <= endIndex && endIndex <= s.length()) :
+            "beginIndex/endIndex must have valid values: 0 <= beginIndex <= endIndex <= length";
+
+        int i = beginIndex;
+        int len = endIndex;
+        int result = 0;
+        while (i < len) {
+            // Accumulating negatively avoids surprises near MAX_VALUE
```

Comment:
I don't get this comment, is it valid? If we want to protect against MAX_VALUE issues we should raise an exception if `digit < 0`, but I don't understand why this is "Accumulating negatively...". Maybe caused by my missing english knowledge.

Reply:
looks like I stripped down more than I should - added few checks just to be on the safe side

Reply:
Could we also adjust the comment? As described I don't understand what is meant with "Accumulating negatively"... or is it only me?

Reply:
ah ok got it, the comment is related to `result -= digit;` the commit is a bit far away :-)